integration/simulation/network/azureenclave.go

```diff
@@ -20,44 +27,104 @@ import (
 
 // creates Obscuro nodes with their own enclave servers that communicate with peers via sockets, wires them up, and populates the network objects
 type networkWithAzureEnclaves struct {
-	ethNodes         []*ethereum_mock.Node
-	obscuroNodes     []*host.Node
+	gethNetwork  *gethnetwork.GethNetwork
+	wallets      []wallet.Wallet
+	contracts    []string
+	workerWallet wallet.Wallet
+
 	obscuroClients   []*obscuroclient.Client
 	enclaveAddresses []string
 }
 
-func NewNetworkWithAzureEnclaves(enclaveAddresses []string) Network {
+func NewNetworkWithAzureEnclaves(enclaveAddresses []string, wallets []wallet.Wallet, workerWallet wallet.Wallet, contracts []string) Network {
 	if len(enclaveAddresses) == 0 {
 		panic("Cannot create azure enclaves network without at least one enclave address.")
 	}
-	return &networkWithAzureEnclaves{enclaveAddresses: enclaveAddresses}
+	return &networkWithAzureEnclaves{
+		enclaveAddresses: enclaveAddresses,
+		wallets:          wallets,
+		contracts:        contracts,
+		workerWallet:     workerWallet,
+	}
 }
 
 func (n *networkWithAzureEnclaves) Create(params *params.SimParams, stats *stats.Stats) ([]ethclient.EthClient, []*obscuroclient.Client, []string, error) {
+	// make sure the geth network binaries exist
+	path, err := gethnetwork.EnsureBinariesExist(gethnetwork.LatestVersion)
+	if err != nil {
+		panic(err)
+	}
+
+	// get wallet addresses to prefund them
+	walletAddresses := make([]string, len(n.wallets))
+	for i, w := range n.wallets {
+		walletAddresses[i] = w.Address().String()
+	}
+
+	// kickoff the network with the prefunded wallet addresses
+	n.gethNetwork = gethnetwork.NewGethNetwork(
+		params.StartPort,
+		params.StartPort+DefaultWsPortOffset,
+		path,
+		params.NumberOfNodes,
+		int(params.AvgBlockDuration.Seconds()),
+		walletAddresses,
+	)
+
+	tmpHostConfig := config.HostConfig{
+		L1NodeHost:          Localhost,
+		L1NodeWebsocketPort: n.gethNetwork.WebSocketPorts[0],
+		L1ConnectionTimeout: DefaultL1ConnectionTimeout,
+	}
+	tmpEthClient, err := ethclient.NewEthClient(tmpHostConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	mgmtContractAddr, err := DeployContract(tmpEthClient, n.workerWallet, common.Hex2Bytes(mgmtcontractlib.MgmtContractByteCode))
+	if err != nil {
+		panic(fmt.Sprintf("failed to deploy management contract. Cause: %s", err))
+	}
+	erc20ContractAddr, err := DeployContract(tmpEthClient, n.workerWallet, common.Hex2Bytes(erc20contract.ContractByteCode))
+	if err != nil {
+		panic(fmt.Sprintf("failed to deploy ERC20 contract. Cause: %s", err))
+	}
+	fmt.Printf("Please start the docker image on the azure server with with:\n")
+	for i := 0; i < len(n.enclaveAddresses); i++ {
+		fmt.Printf("sudo docker run -e OE_SIMULATION=0 --privileged -v /dev/sgx:/dev/sgx -p 11000:11000/tcp obscuro_enclave --hostID %d --address :11000 --managementContractAddress %s  --erc20ContractAddresses %s\n", i, mgmtContractAddr.Hex(), erc20ContractAddr.Hex())
+	}
+	time.Sleep(10 * time.Second)
```

Comment:
I'd be tempted to replace this sleep with a wait like:
```
  fmt.Print("Press 'Enter' when you have started the enclaves...")
  bufio.NewReader(os.Stdin).ReadBytes('\n') 
  ```
if it requires humans to go kick off the image(s). You can interact with that even in Goland console.

Reply:
Fair, I guess devs can just bump the sleep if they need to start more than one or something. I'll do some googling on that goland thing, I'm sure I've used stdin for scripts before in Goland but I think it's getting treated as a test so it's ignoring.